app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -18,37 +19,77 @@
 import com.example.househomey.Filters.MakeFilterFragment;
 import com.example.househomey.Filters.TagFilterFragment;
 import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
 
-public class HomeFragment extends Fragment implements FirestoreUpdateListener {
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This fragment represents the home screen containing the primary list of the user's inventory
+ * @author Owen Cooke, Jared Drueco, Lukas Bonkowski
+ */
+public class HomeFragment extends Fragment {
     private CollectionReference itemRef;
     private ListView itemListView;
     private PopupMenu filterView;
+
+    private ArrayList<Item> itemList = new ArrayList<>();
     private ArrayAdapter<Item> itemAdapter;
 
+    /**
+     * This constructs a new HomeFragment with the appropriate list of items
+     * @param itemRef A reference to the firestore collection containing the items to display
+     */
     public HomeFragment(CollectionReference itemRef) {
         this.itemRef = itemRef;
     }
 
+    /**
+     * @param inflater           The LayoutInflater object that can be used to inflate
+     *                           any views in the fragment,
+     * @param container          If non-null, this is the parent view that the fragment's
+     *                           UI should be attached to.  The fragment should not add the view itself,
+     *                           but this can be used to generate the LayoutParams of the view.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed
+     *                           from a previous saved state as given here.
+     * @return the home fragment view containing the inventory list
+     */
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         // Inflate the fragment's layout
         View rootView = inflater.inflate(R.layout.fragment_home, container, false);
 
         itemListView = rootView.findViewById(R.id.item_list);
-        itemAdapter = new ItemAdapter(getContext(), new ItemList(this, itemRef).getItems());
+        itemAdapter = new ItemAdapter(getContext(), itemList);
         itemListView.setAdapter(itemAdapter);
 
+        itemRef.addSnapshotListener((querySnapshots, error) -> {
```

Comment:
@ldbonkowski is it cool if we put this in its own method? For example `setupItemListener`. Just to keep things a bit more modular and potentially reusable.

Reply:
Yessit, good plan